odoo/addons/base/models/avatar_mixin.py

```diff
@@ -0,0 +1,110 @@
+    _name = 'avatar.mixin'
+    _description = "Avatar Mixin"
+
+    # all image fields are base64 encoded and PIL-supported
+
+    avatar_1920 = fields.Image("Image", max_width=1920, max_height=1920, compute="_compute_avatar_1920", inverse="_set_avatar", readonly=False, search="_avatar_1920_search")
+
+    avatar_1024 = fields.Image("Image 1024", max_width=1024, max_height=1024, compute="_compute_avatar_1024", readonly=False)
+    avatar_512 = fields.Image("Image 512", max_width=512, max_height=512, compute="_compute_avatar_512", readonly=False)
+    avatar_256 = fields.Image("Image 256", max_width=256, max_height=256, compute="_compute_avatar_256", readonly=False)
+    avatar_128 = fields.Image("Image 128", max_width=128, max_height=128, compute="_compute_avatar_128", readonly=False)
+
+    # image fields are stored and hold the image data
+
+    image_1920 = fields.Image("Image Placeholder", max_width=1920, max_height=1920, store=True)
+
+    # resized fields stored for performance
+    image_1024 = fields.Image("Image Placeholder 1024", related="image_1920", max_width=1024, max_height=1024, store=True)
+    image_512 = fields.Image("Image Placeholder 512", related="image_1920", max_width=512, max_height=512, store=True)
+    image_256 = fields.Image("Image Placeholder 256", related="image_1920", max_width=256, max_height=256, store=True)
+    image_128 = fields.Image("Image Placeholder 128", related="image_1920", max_width=128, max_height=128, store=True)
+
+    @api.depends(lambda self: [self._rec_name, 'avatar_1920', 'image_1920'])
+    def _compute_avatar_1920(self):
+        for record in self:
+            record['avatar_1920'] = record['image_1920'] or generate_avatar_svg(record.name)
+
+    @api.depends(lambda self: [self._rec_name, 'avatar_1024', 'image_1024'])
+    def _compute_avatar_1024(self):
+        for record in self:
+            record['avatar_1024'] = record['image_1024'] or generate_avatar_svg(record.name)
+
+    @api.depends(lambda self: [self._rec_name, 'avatar_512', 'image_512'])
+    def _compute_avatar_512(self):
+        for record in self:
+            record['avatar_512'] = record['image_512'] or generate_avatar_svg(record.name)
+
+    @api.depends(lambda self: [self._rec_name, 'avatar_256', 'image_256'])
+    def _compute_avatar_256(self):
+        for record in self:
+            record['avatar_256'] = record['image_256'] or generate_avatar_svg(record.name)
+
+    @api.depends(lambda self: [self._rec_name, 'avatar_128', 'image_128'])
+    def _compute_avatar_128(self):
+        for record in self:
+            record['avatar_128'] = record['image_128'] or generate_avatar_svg(record.name)
+
+    def _set_avatar(self):
+        for record in self:
+            record['image_1920'] = record['avatar_1920']
+
+    def _avatar_1920_search(self, operator, value):
+        return [('image_1920', operator, value)]
+
+def _encode_name(name):
+    return sha512(name.encode('utf-8'))
```

Comment:
Shouldn't this actually depend on id instead of name? Or a combination of both?

I would be hoping that 2 persons that have the same name can at least be distinguished with a different avatar colors.

Reply:
Also if someone is changing his name for some reason, I would hope he keeps the same color actually, so just using id sounds better?

Reply:
And if we want this to be export/import proof as well, maybe `create_date` or something similar instead of `id`?

Reply:
`create_date` was being used before and we had a huge problem because employees `create_date` were different as the users `create_date`. So the employee `mitchell admin` would have a different color than the user `mitchell admin`.

Reply:
Then I think all of those methods should be defined on the mixin itself, so that each model can override them as they see fit.

And user would override this one to return the value from its corresponding partner instead.

Reply:
When speaking to the PO, he said it would be ok to have the same color for two users with the same name to have the same avatar. I think if it's easy to fix, we should do it, but I don't see a way other than using the id to compute the hash

Reply:
As said just before, I think now all of those methods should be part of the mixin itself, so that they can be overridden per model if necessary.

And override can be done so that partner and user have the same placeholder, and also employee and user if applicable, in a way that would work with the create_date if possible then.